lib/main_common.pm

```diff
@@ -86,6 +88,13 @@ sub setup_env {
     }
 }
 
+sub any_desktop_is_applicable() {
```

Comment:
is_desktop_installed - get_var("DESKTOP") !~ /textmode|minimalx/;

I actually also not very happy about both functions , their naming and whole idea behind. But I think that it's really bad idea in terms of one PR which suppose to solve problem A try to also problem B just because it touch same code. I didn't create this functions I just move them from one file to another.